### PR TITLE
fix Issue with boss killscore

### PR DESCRIPTION
**Pull request by S-T-A-L-K-R**
### Brief description of the changes:
Attempt to fix issue "Boss killed doesnt count as killscore? #346". Yes, big part of them doesn't count.

### Tested Changes:
- [X] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

I've tested it locally, and despite
```
 262.357 Script @C:/Users/{User}/AppData/Roaming/Factorio/temp/currently-playing/utils/event_core.lua:51: [ERROR] Given force is not valid, it must be a valid force index, a string (name of the force) or a LuaForce. in .../Factorio/temp/currently-playing/functions/boss_unit.lua:31
```
it seems to be working.

### CODE REVIEW

File: functions/boss_unit.lua [hunk 1]
```diff
@@ -38,7 +38,7 @@ local function on_entity_damaged(event)
 	boss.health = boss.health - event.final_damage_amount
 	if boss.health <= 0 then
 		global.boss_units[entity.unit_number] = nil
-		entity.die()
+		entity.die(event)
```
Comment (TheBigZet) [inline functions/boss_unit.lua:41]: According to the API https://lua-api.factorio.com/latest/LuaEntity.html#LuaEntity.die `die()` can take `LuaForce` and `LuaEntity` as a cause. I doubt your PR fixes anything. Perhaps passing event.force and/or event.cause would do sth

**Review by TheBigZet — CHANGES_REQUESTED** [review]

**Comment by S-T-A-L-K-R** [comment]
Factorio API wasn't as helpful as i hoped.
I randomly deleted manual invocation of "entity.die()" and it works, now even without errors in console.
I believe manual "entity.die()" call, caused "on_entity_died()" to not receive proper data.
Evidence: https://youtu.be/t4u_NaFJonY
Console final output(because i forgot to record it): 
FactorioConsoleDump

**Comment by TheBigZet** [comment]
Did you check that the score is in fact increasing?
`on_entity_died()` has to recive at least `cause` https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/blob/master/comfy_panel/score.lua#L388

Removing `entity.die()` will result in boss becoming normal biter after his healthbar gies to 0. Which makes then stronger and changes healthbar more into shield

**Comment by S-T-A-L-K-R** [comment]
yea, passing _entity.cause_ as a force seems to be working better

**Comment by clifffrey** [comment]
This has been fixed by #436

**Pull request closed without merging**